crates/wasm-smith/src/config.rs

```diff
@@ -429,6 +429,7 @@ pub struct SwarmConfig {
     pub available_imports: Option<Vec<u8>>,
     pub bulk_memory_enabled: bool,
     pub canonicalize_nans: bool,
+    pub component_model: bool,
```

Comment:
Sorry I forgot that we don't actually want to create this `bool` flag because nothing inside `wasm-smith` will ever use it in practice since we statically know whether we are doing the component model or not based on whether we call into `wasm-smith` via `Module::new` or via `Component::new`.

So let's remove this flag, and instead...